w3-session.md

```diff
@@ -0,0 +1,210 @@
+| Authority   | An agent that interprets a UCAN and verifies that it is valid                                                                                  |
+| Session key | The [`did:key`] identifier of the key that is used to sign UCANs issued by Identity. Usually this is a DID of the Agent authorized as Identity |
+
+## Premise
+
+Mapping cryptographic [`did:key`] identifier to an identity with memorable identifier like [`did:mailto`] MUST be authorized by the identity owner. When identity is identified with a a non-cryptographic identifier authorization is not self-evident from the UCAN delegation and requires performing out-of-bound authentication like [DID resolution] or an email based confirmation flow.
+
+To keep UCANs verifiable in stateless settings we propose capturing stateful information (out-of-bound authorization) in a stateless [session], represented using a UCAN, so it can be included in verifiable proofs.
+
+### Authorization
+
+An agent MAY request an authorization [session] by invoking `access/authorize` capability that specifies desired identity (`nb.as`) and session key (`with`).
+Capability provider (notary) MUST carry out-of-bound authentication that allows identity owner to accept or deny authorization request.
+
+```mermaid
+sequenceDiagram
+  participant Agent as 👩‍💻<br/><br/>did:key:zAgent
+  participant W3 as 🌐<br/><br/>did:web:web3.storage #32;
+  participant Email as 📬<br/><br/>alice@web.mail
+
+  Agent ->> W3: access/authorize
+  Note right of Agent:🎟<br/>with: did:key:zAgent<br/>as: did:mailto:alice@web.mail
+  W3 ->> Email: ✉️ Verification email
+  Email ->> W3: 🔗 Approve
+  W3 -->> Agent: ./update
+    Note right of Agent:🎫<br/>with: did:web:web3.storage<br/>key: did:key:zAgent
+```
+
+#### Authorization Example
+
+> Request authorization from `alice@web.mail` with `did:key:zAgent`.
+
+```ts
+{
+  "iss": "did:key:zAgent",        // agent
+  "aud": "did:web:web3.storage",  // notary
+  "att": [{
+    "with": "did:key:zAgent",     // session key
+    "can": "access/authorize",
+    "nb": { as: "did:mailto:web.mail:alice" }  // identity
+  }]
+}
+```
+
+#### Authorization Principal
+
+The `nb.as` field MUST specify a principal that requesting agent wishes to represent in authorized session. It MUST be valid [UCAN principal] so that it can be used as issuer (`iss`) in UCANs covered by authorized session.
+
+#### Authorization Key
+
+Resource (`with`) MUST be a [`did:key`][] identifier for the desired session key. It represents a key that agent wishes to use for signing [UCAN][]s that it will issue using authorization principal.
+
+> Please note that session key `with`, MAY be different or same as the issuing agent DID (`iss`).
+
+### Authentication
+
+Agent providing `access/authorize` capability (notary) MUST carry out-of-bound process in which authorizing principal may confirm or deny authorization. Process MAY vary based on type of identity.
```

Comment:
> based on type of identity

Does this mean something different than 'based on DID method'?

Reply:
Probably no, but perhaps there will be a case. I don't think it's worth committing to that exactly.